File: src/2022/day10/src/main/java/ca/terrylockett/aoc2022/day10/Day10Runner.java
package ca.terrylockett.aoc2022.day10;

import ca.terrylockett.aoccommon.resources.Resources;

public class Day10Runner {

	public static void main(String[] args) throws Exception {
		String inputFile = Resources.getInputFilePath("input.txt").orElseThrow();

		// part 1
		ClockCircuit clock = new ClockCircuit();
		clock.procesInstructionSet(inputFile);
		System.out.println("Day 10 Part 01: " + clock.calcSignalStrengthsSum());

		// part 2
		System.out.println("Day 10 Part 02: ");
		for (int i = 0; i < 6; i++) {
			System.out.println(clock.renderCrtLine(i));
		}
	}
}

File: src/2023/day10/src/main/kotlin/ca/terrylockett/aoc2023/day10/Day10Runner.kt
package ca.terrylockett.aoc2023.day10;

import ca.terrylockett.aoccommon.resources.Resources;

fun main() {
    val inputFile: String = Resources.getInputFilePath("input.txt").orElseThrow()
    println("2023 day10 part1: ${part1(inputFile)}")
    //println("2023 day10 part2: TODO")
}


fun part1(inputFilePath: String) : Int {
    val pipeMap = PipeMap(inputFilePath)
    return pipeMap.traverse()
}
    



